lang/cs/Org.Apache.REEF.Client/YARN/YARNREEFClient.cs

```diff
@@ -123,42 +124,76 @@ private void Launch(IJobSubmission jobSubmission, string driverFolderPath)
 
             // TODO: Remove this when we have a generalized way to pass config to java
             var paramInjector = TangFactory.GetTang().NewInjector(jobSubmission.DriverConfigurations.ToArray());
-                
-            var avroJobSubmissionParameters = new AvroJobSubmissionParameters
+
+            var submissionJobArgsFilePath = SerializeJobFile(jobSubmission, driverFolderPath);
+            var submissionAppArgsFilePath = SerializeAppFile(jobSubmission, paramInjector, driverFolderPath);
+
+            // Submit the driver
+            _javaClientLauncher.Launch(JavaClassName, submissionJobArgsFilePath, submissionAppArgsFilePath);
+            Logger.Log(Level.Info, "Submitted the Driver for execution." + jobSubmission.JobIdentifier);
+        }
+
+        private string SerializeAppFile(IJobSubmission jobSubmission, IInjector paramInjector, string driverFolderPath)
```

Comment:
The two different submission clients use different parameters and different return types, I'm not convinced that this will generalize well as an interface.